### DISCUSSION TITLE
Intermittent error, 22P02 - input string ended unexpectedly

### DISCUSSION THREAD

**Discussion by ricgagliardi** [supabase/supabase#1254] [Questions] [answered]
I have a number of HTTP requests that run fine most of the time, but sometimes return an error:

    code: 22P02
    details: The input string ended unexpectely
    message: invalid input format for type json

I have a couple of examples below. You'll see that the http requests are identical, yet sometimes they work and sometimes they fail. It seems they fail in batches, I may go hours without an error, then get a batch of 10 of them.

This is for a volunteer project sponsored by the League of Women Voters. We chose Supabase for the speed at which I could develop the app (a game for high school students). Our rollout is planned for the first week of May, but this problem is a show stopper for us. I haven't been able to find any work-around.

*Does anyone see a problem? Or is this a problem within Supabase?*

Here are a couple of examples:

#### SUCCESS: 
https://cqxvknmoofzuhxbonmoj.supabase.co/rest/v1/play
Request Method: POST
{"game":"News","play_code":-1}

returns: 201 
[{"play":187,"game":"News","play_code":-1}]

#### FAILURE:
https://cqxvknmoofzuhxbonmoj.supabase.co/rest/v1/play
Request Method: POST
{"game":"News","play_code":-1}

returns: 400 
{"hint":null,"details":"The input string ended unexpectedly.","code":"22P02","message":"invalid input syntax for type json"}

#### SUCCESS:
https://cqxvknmoofzuhxbonmoj.supabase.co/rest/v1/source?or=(name.ilike.Mediabiasfactcheck.com,name.ilike.Constitutioncenter.org)
returns: 200
[{"name":"Mediabiasfactcheck.com","description": ... ,"hints":null}]
(note, response edited for brevity)

#### FAILURE:
https://cqxvknmoofzuhxbonmoj.supabase.co/rest/v1/source?or=(name.ilike.Mediabiasfactcheck.com,name.ilike.Constitutioncenter.org)

returns: 400
{"hint":null,"details":"The input string ended unexpectedly.","code":"22P02","message":"invalid input syntax for type json"}

**Comment by steve-chavez** [comment, marked as answer]
> FAILURE:
https://cqxvknmoofzuhxbonmoj.supabase.co/rest/v1/play
Request Method: POST
{"game":"News","play_code":-1}

> returns: 400
{"hint":null,"details":"The input string ended unexpectedly.","code":"22P02","message":"invalid input syntax for type json"}

That looks like some of your rows might have invalid json in their values. Can you confirm this is the full data POSTed? 

It could be that be some values have un-escaped double quotes(`"`), for example.

**Reply by ricgagliardi** [reply]
Thanks for the idea, but that can't be the problem. The **exact same query** works correctly most of the time, sending the same data. When it randomly starts to happen, I may have to refresh the page 4-5 or more times, and the next time the request succeeds. No data is changed on the back-end in the meantime.  Same for the POST. And it happens on other requests too.

Also, I know that GET request looks pretty bizarre. I started with simpler versions and tried that one as a work-around. It's just the last version I tried when I posted this.

**Reply by inian** [reply]
Hey Ric, can you email beta@supabase.io? I tried reaching out with the email we have on record, but there seems to be some issue.

**Reply by ricgagliardi** [reply]
Thanks, I just did that. I'll also check on the problem you have with my email address on record.

**Comment by farzd** [comment]
i'm having this issue in production as well. Randomly. Haven't updated my code. And it actually enters the data into the db, but returns the error

my payload
```
[
    {
        "completed": false,
        "prizes": [],
        "type": "daily",
        "url": "https://youtu.be/6Ypk95FaOFQ",
        "startDate": "2022-06-17T02:03:00.000Z",
        "endDate": "2022-06-18T02:03:00.000Z",
        "createdBy": "910b7452-7532-4ee2-9168-334bf6123cd6",
        "entry": null
    }
]
```

**Reply by steve-chavez** [reply]
This was related to https://github.com/supabase/supabase/issues/7293, basically a cloudflare error.
